Create urh_fec_decode_cc1101.cpp

SpaceTeddy opened the pull request:
external decoding plugin for CC1101 FEC (Forward Error Correction) decoding in urh.
simply compile with: g++ urh_fec_decode_cc1101.cpp -o urh_fec_decode_cc1101

start the plugin in urh as external program with: urh_fec_decode_cc1101 [d | f]

Parameter description: 
d = output decoded payload incl. preamble and sync word. 
note: preamble and sync word are not scrambled 

f = output decoded payload, but preamble and sync word are cropped from the data stream

![image](https://cloud.githubusercontent.com/assets/9148964/26417496/a91d0f9c-40b9-11e7-9a54-ae8ea780f551.png)


https://github.com/SpaceTeddy/urh-cc1101_FEC_decode_plugin

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/jopohl/urh?pullRequest=276) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/jopohl/urh?pullRequest=276) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/jopohl/urh?pullRequest=276) it.</sub>

andynoack commented:
Hi @SpaceTeddy,

thank you for your contribution. We discussed the pull request and think that it is not the right way to integrate your work.
1. Every time you update/fix the code we have to fix the project code accordingly.
2. The current state of your implementation does not allow to encode data which is a must have for our decodings as long as this is possible (e.g. not for cut obviously).

We would like to integrate it in the following way: Our documentation will have a table for external extensions and plugins (or an explicit repo). There we can link your github project. What do you think?

SpaceTeddy commented:
This decoding tool is just an offer and there is no need to merge it into your repository. I don't think that I will do some updates in this tool.
From my point of view you can reject the request. I'm not angry if it don't fit to your strategy 

Regards,
Chris

vsboost commented:
@SpaceTeddy Thanks for your plugin.

SpaceTeddy commented:
@andynoack ,
I've added FEC encoding support to the plugin and changed the name of the repository to
[urh-cc1101_FEC_encode_decode_plugin](https://github.com/SpaceTeddy/urh-cc1101_FEC_encode_decode_plugin)
The new version is now able to encode a non FEC encoded payload.

start the plugin in urh as external program with: 
urh_fec_decode_cc1101 [e | d | f]

Parameter description: 
e = encodes a cc1101 FEC unencoded payload 
d = output decoded payload incl. preamble and sync word. note: preamble and sync word are not scrabled 
f = output decoded payload, but preamble and sync word are cropped from the data stream

andynoack commented:
Great! Thanks! We are preparing something to include your submission but at the moment time is rare...sorry for that...it is on our todo list!

jopohl commented:
Hi @SpaceTeddy 
thanks for your work and the Readme you provided in the Repo. I created a page in the wiki so others can find it: https://github.com/jopohl/urh/wiki/External-Decodings-ready-for-use
